src/Decompiler/Typing/TypedConstantRewriter.cs

```diff
@@ -365,6 +363,39 @@ DataType PromoteToCString(Constant c, DataType charType)
             return dt;
         }
 
+        private bool IsInsideField(int offset, StructureField field)
+        {
+            if (offset == field.Offset)
+                return true;
+            if (offset < field.Offset + field.DataType.Size)
+                return true;
+            var str = field.DataType.ResolveAs<StructureType>();
```

Comment:
Should this not be:
```
if (field.Offset <= offset && offset < field.Offset + field.DatType.Size)
```

Reply:
`StructureFieldCollection.LowerBound`  guarantees that `field.Offset <= offset` is always `true`. But I'll add additional check for safety